contracts/Curation.sol

```diff
@@ -0,0 +1,415 @@
+            // it's still in application phase (not registered yet)
+            if (!application.registered) {
+                // insert in Registry app
+                registry.add(application.data);
+                application.registered = true;
+            }
+            // Remove applicant (winner) used lock
+            delete(usedLocks[application.applicant][application.lockId]);
+            // Unlock challenger tokens from Staking app
+            reward = challenge.amount.mul(dispensationPct) / PCT_BASE;
+            // Redistribute tokens
+            staking.unlockAndMoveTokens(challenge.challenger, challenge.lockId, application.applicant, reward);
+        } else { // challenge accepted (application rejected)
+            // it has been already registered
+            if (application.registered) {
+                // remove from Registry app
+                registry.remove(entryId);
+                application.registered = false;
+            }
+            // Remove challenger (winner) used lock
+            delete(usedLocks[challenge.challenger][challenge.lockId]);
+            // Unlock applicant tokens from Staking app
+            reward = application.amount.mul(dispensationPct) / PCT_BASE;
+            // Redistribute tokens
+            staking.unlockAndMoveTokens(application.applicant, application.lockId, challenge.challenger, reward);
+        }
+
+        ResolvedChallenge(entryId, voteResult);
+    }
+
+    function claimReward(bytes32 entryId) isInitialized public {
+        require(isChallengeResolved(entryId));
+
+        Challenge storage challenge = challenges[entryId];
+        Application memory application = applications[entryId];
+
+        // avoid claiming twice
+        require(!challenge.claims[msg.sender]);
+
+        // register claim to avoid claiming it again
+        challenge.claims[msg.sender] = true;
+
+        bool voteResult;
+        uint256 totalWinningStake;
+        (voteResult, totalWinningStake,) = voting.getVoteResult(challenge.voteId);
+
+        address loser;
+        uint256 loserLockId;
+        uint256 amount;
+        if (voteResult == false) {
+            loser = challenge.challenger;
+            loserLockId = challenge.lockId;
+            amount = challenge.amount;
+        } else { // voteResult == true
+            loser = application.applicant;
+            loserLockId = application.lockId;
+            amount = application.amount;
+        }
+
+        // reward as a voter
+        uint256 voterWinningStake = voting.getVoterWinningStake(challenge.voteId, msg.sender);
+        require(voterWinningStake > 0);
+        // amount * (voter / total) * (1 - dispensationPct)
+        uint256 reward = amount.mul(voterWinningStake).mul(PCT_BASE.sub(dispensationPct)) / (totalWinningStake * PCT_BASE);
+        // Redistribute tokens
+        staking.unlockAndMoveTokens(loser, loserLockId, msg.sender, reward);
```

Comment:
I think it would be cleaner if the curation app just took all the tokens from the challenger or applicant on`resolveChallenge(...)`. This way we don't need to modify the lock every time, being more efficient.

Also in terms of the user that lost some tokens, I think it is better to remove them from their account ASAP as there is no way they are going to recover them.

Reply:
Completely agreed. Plus:
- It would solve the `TODO` 3 lines below about that truncating issue.
- It follows the same logic as our [PLCR](https://github.com/aragonlabs/plcr/blob/contracts_first_version/contracts/PLCR.sol#L107)